src/v/iceberg/manifest_io.cc

```diff
@@ -53,7 +53,7 @@ manifest_io::download_manifest_list(const manifest_list_path& path) {
 ss::future<checked<size_t, metadata_io::errc>>
 manifest_io::upload_manifest(const manifest_path& path, const manifest& m) {
     return upload_object<manifest>(
-      path(), m, "iceberg::manifest", [](const manifest& m) {
+       from_uri(path().string()), m, "iceberg::manifest", [](const manifest& m) {
```

Comment:
nit: it would be great if we could maybe make a separate method fro URIs. Before this change, it was somewhat clear that a manifest_path was not a URI but a raw path. With this change it's a bit more ambiguous

Reply:
I think it'd be helpful to also rename get_manifest_list_path() to get_manifest_list_uri() or something similar